mongo/integration/server_selection_prose_test.go

```diff
@@ -21,9 +23,80 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+// ? do we want to extend saturation set as an mtest.T method?
+
+// saturationSet is used to maintain information about events with specific host+pool combinations.
+type saturationSet map[string]bool
```

Comment:
Is it okay that we decouple this from the sub-tests?  Do we want to make it part of the `mtest` package?  It's the code around which we ensure saturation of all pools+hosts within a client.